ci/checks/check_min_airflow_dependency_compatibility.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Install great_expectations using the airflow constraints file for our minimum supported version
+
+AIRFLOW_VERSION=2.2.3
```

Comment:
@anthonyburdi @talagluck 
Do we say anything about the minimum version of airflow that we support?

Reply:
We support 2.x.latest. So we would want to test with the latest patch version of each minor version starting with 2.0.latest.

Reply:
Supporting `2.0` might require more work than I what I've done in this PR.
I'll add a comment about this and lower it as much as possible, but going much lower will likely require a followup.

Reply:
I'd like to know how low we can go with this PR, and lock it there

Reply:
Going update this comment with all the dependency conflicts or errors we get.

### `2.0.2 -> 2.1.4`
We have a conflict with `pytz` which prevents installation.
```
ERROR: Cannot install great-expectations==0.16.3+55.g57750997f.dirty because these package versions have conflicting dependencies.

The conflict is caused by:
    great-expectations 0.16.3+55.g57750997f.dirty depends on pytz>=2021.3
    The user requested (constraint) pytz==2021.1
```


### `2.2.0`
We can install but we fail on `get_context()`
```
Traceback (most recent call last):
  File "ci/checks/get_context.py", line 1, in <module>
    import great_expectations as gx
  File "/usr/local/lib/python3.8/site-packages/great_expectations/__init__.py", line 6, in <module>
    from great_expectations.data_context.migrator.cloud_migrator import CloudMigrator
  File "/usr/local/lib/python3.8/site-packages/great_expectations/data_context/__init__.py", line 1, in <module>
    from great_expectations.data_context.data_context import (
  File "/usr/local/lib/python3.8/site-packages/great_expectations/data_context/data_context/__init__.py", line 1, in <module>
    from great_expectations.data_context.data_context.abstract_data_context import (
  File "/usr/local/lib/python3.8/site-packages/great_expectations/data_context/data_context/abstract_data_context.py", line 52, in <module>
    from great_expectations.core.config_provider import (
  File "/usr/local/lib/python3.8/site-packages/great_expectations/core/config_provider.py", line 9, in <module>
    from great_expectations.core.config_substitutor import _ConfigurationSubstitutor
  File "/usr/local/lib/python3.8/site-packages/great_expectations/core/config_substitutor.py", line 22, in <module>
    import boto3
  File "/usr/local/lib/python3.8/site-packages/boto3/__init__.py", line 16, in <module>
    from boto3.session import Session
  File "/usr/local/lib/python3.8/site-packages/boto3/session.py", line 17, in <module>
    import botocore.session
  File "/usr/local/lib/python3.8/site-packages/botocore/session.py", line 30, in <module>
    import botocore.credentials
  File "/usr/local/lib/python3.8/site-packages/botocore/credentials.py", line 34, in <module>
    from botocore.config import Config
  File "/usr/local/lib/python3.8/site-packages/botocore/config.py", line 16, in <module>
    from botocore.endpoint import DEFAULT_TIMEOUT, MAX_POOL_CONNECTIONS
  File "/usr/local/lib/python3.8/site-packages/botocore/endpoint.py", line 22, in <module>
    from botocore.awsrequest import create_request_object
  File "/usr/local/lib/python3.8/site-packages/botocore/awsrequest.py", line 26, in <module>
    import botocore.utils
  File "/usr/local/lib/python3.8/site-packages/botocore/utils.py", line 33, in <module>
    import botocore.httpsession
  File "/usr/local/lib/python3.8/site-packages/botocore/httpsession.py", line 18, in <module>
    from urllib3.contrib.pyopenssl import orig_util_SSLContext as SSLContext
  File "/usr/local/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py", line 50, in <module>
    import OpenSSL.SSL
  File "/usr/local/lib/python3.8/site-packages/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import SSL, crypto
  File "/usr/local/lib/python3.8/site-packages/OpenSSL/SSL.py", line 19, in <module>
    from OpenSSL.crypto import (
  File "/usr/local/lib/python3.8/site-packages/OpenSSL/crypto.py", line 3258, in <module>
    utils.deprecated(
TypeError: deprecated() got an unexpected keyword argument 'name'

```